build-logic/src/main/kotlin/org/jetbrains/conventions/dokka-base-frontend-files.gradle.kts

```diff
@@ -0,0 +1,32 @@
+package org.jetbrains.conventions
+
+import org.gradle.api.attributes.Usage.USAGE_ATTRIBUTE
+
+/**
+ * Utility for sharing the Dokka Base Plugin frontend files between subprojects in a safe, cacheable way.
+ */
+
+plugins {
+    id("org.jetbrains.conventions.base")
+}
+
+/** Apply a distinct attribute to the incoming/outgoing configuration */
+fun AttributeContainer.dokkaBaseFrontendFilesAttribute() =
+    attribute(USAGE_ATTRIBUTE, objects.named("org.jetbrains.dokka.base-frontend-files"))
```

Comment:
I would propose to rename all mentions of `base-frontend-files` to `html-frontend-files` while we're here.

The HTML plugin (and thus the format itself) will one day be extracted out of `base` and into it's own module, it's already been attempted in #2752, but we switched to more urgent matters

This convention plugin also has great potential for reusability, and in theory the frontend files could reside in multiple other modules that have nothing to do with `base`, so to me it makes sense to rename it to something broader

Reply:
> I would propose to rename all mentions of `base-frontend-files` to `html-frontend-files` while we're here.

Good idea, done!

By the way, now that the Gradle config of Dokka is more Gradle-idiomatic (e.g. there's no `allprojects {}` Gradle configuration, subprojects are mostly isolated), it might help to think about restructuring the subprojects while you're doing this.

Personally I think the subproject nesting and terse subproject names can be confusing, so I prefer a more flat layout so that all of the subprojects are visible at once, while the subproject names are expanded so they are ordered. So something like this:

```
.
├── dokka-core/
│   ├── dokka-generator
│   ├── dokka-generator-configuration
│   ├── dokka-content-matcher-test-utils
│   └── dokka-test-api
└── dokka-plugins/
    ├── dokka-html
    ├── dokka-html-frontend
    ├── dokka-gfm
    ├── dokka-gfm-template-processing
    └── ...
```

(this example includes theoretical splitting of subprojects

The [Kotest](https://github.com/kotest/kotest/) and [Gradle](https://github.com/gradle/gradle/tree/master/subprojects) projects are examples of this layout that I like.

Just sharing an idea :)

Reply:
Good suggestion for sure, there's definitely lots of room for improvement! 

Although there's a bit too much refactoring going on at the moment with K2 and the extraction of the html format out of dokka-base, it'll introduce a lot nasty merge conflicts for several long-lived branches... Should be easier to do it closer to ~1.9.20 when as much as possible gets merged into master though